src/libraries/System.Net.Mail/tests/Unit/MailAddressTests/MailAddressDisplayNameTest.cs

```diff
@@ -21,6 +22,45 @@ public void MailAddress_WithUnicodeDisplayAndMailAddress_ToStringShouldReturnDis
             Assert.Equal(string.Format("\"{0}\" <{1}>", DisplayNameWithUnicode, Address), _mailAddress.ToString());
         }
 
+        [Theory]
+        [InlineData(Address, "test\"Display\"Name")]
+        [InlineData(Address, "Hello \"world hello\" world")]
+        [InlineData(Address, "Hello \"world")]
+        [InlineData(Address, "Hello \"\"world")]
+        [InlineData(Address, "\"")]
+        [InlineData(Address, "Hello \\\"world hello\\\" world")]
+        public void MailAddress_WithDoubleQuotesDisplayAndMailAddress_ToStringShouldReturnDisplayNameEscapeSequenceAndAddressInAngleBrackets(string address, string displayNameWithDoubleQuotes)
+        {
+            MailAddress mailAddress = new MailAddress(address, displayNameWithDoubleQuotes);
+            Assert.Equal(displayNameWithDoubleQuotes, mailAddress.DisplayName);
+            Assert.Equal(string.Format("\"{0}\" <{1}>", displayNameWithDoubleQuotes.Replace("\"", "\\\""), Address), mailAddress.ToString());
+        }
+
+        [Theory]
+        [InlineData(Address, "\"John Doe\"")]
+        [InlineData(Address, "\"\"")]
+        public void MailAddress_WithDoubleQuotesInBeginningAndInEndingDisplayAndMailAddress_ToStringShouldReturnDisplayNameEscapeSequenceAndAddressInAngleBrackets(string address, string displayNameWithDoubleQuotesInBeginningAndInEnding)
+        {
+            if (displayNameWithDoubleQuotesInBeginningAndInEnding.Length >= 2 && displayNameWithDoubleQuotesInBeginningAndInEnding[0] == '\"' && displayNameWithDoubleQuotesInBeginningAndInEnding[^1] == '\"')
```

Comment:
You don't need the test here, the test is only for such data.

Reply:
as you told me to split the test cases so i created separate test case for kind of test what would be recommended action here ?

Reply:
In the comment, I mean the `if` since you already know it starts and ends with '"' by the definition of the test.

Reply:
sorry i had internet issues and i was not able to review your comment properly , but now i have applied the all suggestion you told , please review it.

Reply:
Looks good now, Karel have to approve as well though. Thanks for the patience and your contribution :+1:

Reply:
@ManickaP @karelz @wfurt i really appreciate all your feedback and valuable opinion and kind help, since I was new to github it self this is the first contribution I did in lifetime so I was   kind of newbie but all your guidance and help made worth of contributing and really looking forward to do more  love 😍 .net